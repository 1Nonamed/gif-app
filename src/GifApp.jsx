import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

const GifApp = () => {
  const [categories, setCategories] = useState(['Steins Gate']);

  const onAddCategory = (newCategory) => {
    console.log(newCategory);
    if (categories.includes(newCategory)) return;

    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GifApp</h1>
      <AddCategory onNewCategory={onAddCategory} />

      {categories?.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};

export default GifApp;
