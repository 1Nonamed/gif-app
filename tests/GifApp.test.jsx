import { fireEvent, render, screen } from '@testing-library/react';
import GifApp from '../src/GifApp';

describe('Tests on <GifApp />', () => {
  test('should match the snapshot', () => {
    const { container } = render(<GifApp />);

    expect(container).toMatchSnapshot();
  });

  test('should show a heading h3 with "GifApp"', () => {
    render(<GifApp />);

    const h1 = screen.getByRole('heading', { level: 1 });

    expect(h1.innerHTML).toBe('GifApp');
  });

  test('should show a new category', () => {
    render(<GifApp />);

    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form');

    fireEvent.input(input, { target: { value: 'One Piece' } });
    fireEvent.submit(form);

    expect(screen.getByText('One Piece')).toBeTruthy();
  });

  test('should not add an existing category', () => {
    render(<GifApp />);

    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form');

    // Steins Gate is my default value
    fireEvent.input(input, { target: { value: 'Steins Gate' } });
    fireEvent.submit(form);

    expect(screen.getAllByRole('heading', { level: 3 })).toHaveLength(1);
  });
});
