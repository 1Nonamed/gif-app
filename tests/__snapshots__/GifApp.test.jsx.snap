// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifApp /> should match the snapshot 1`] = `
<div>
  <h1>
    GifApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Steins Gate
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
